pkg/scheduler/scheduler.go

```diff
@@ -771,26 +759,32 @@ func patchClusterBindingStatusCondition(karmadaClient karmadaclientset.Interface
 	if newScheduledCondition.Status == metav1.ConditionTrue {
 		updateCRB.Status.SchedulerObservedGeneration = crb.Generation
 	}
+
+	if reflect.DeepEqual(crb.Status, updateCRB.Status) {
+		return nil
+	}
 	return patchClusterResourceBindingStatus(karmadaClient, crb, updateCRB)
 }
 
 // patchClusterBindingStatusWithAffinityName patches schedule status with affinityName of ClusterResourceBinding when necessary.
 func patchClusterBindingStatusWithAffinityName(karmadaClient karmadaclientset.Interface, crb *workv1alpha2.ClusterResourceBinding, affinityName string) error {
+	if crb.Status.SchedulerObservedAffinityName == affinityName {
+		return nil
+	}
+
 	klog.V(4).Infof("Begin to patch status with affinityName(%s) to ClusterResourceBinding(%s).", affinityName, crb.Name)
 	updateCRB := crb.DeepCopy()
 	updateCRB.Status.SchedulerObservedAffinityName = affinityName
 	return patchClusterResourceBindingStatus(karmadaClient, crb, updateCRB)
 }
 
 func patchClusterResourceBindingStatus(karmadaClient karmadaclientset.Interface, crb, updateCRB *workv1alpha2.ClusterResourceBinding) error {
-	// Short path, ignore patch if no change.
-	if reflect.DeepEqual(crb.Status, updateCRB.Status) {
-		return nil
-	}
-
-	patchBytes, err := helper.GenMergePatch(crb, updateCRB)
+	patchBytes, ok, err := helper.GenMergePatch(crb, updateCRB)
 	if err != nil {
-		return fmt.Errorf("failed to create a merge patch: %v", err)
+		return err
+	}
+	if !ok {
```

Comment:
If we want to check whether a patch is empty or not, why not just use `if len(patchBytes) == 0`? It's more straightforward I guess.

Reply:
Thanks for your suggestion.